frontend/packages/ceph-storage-plugin/src/components/modals/clone-pvc-modal/clone-pvc-modal.tsx

```diff
@@ -0,0 +1,174 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Form, FormGroup, TextInput } from '@patternfly/react-core';
+import {
+  convertToBaseValue,
+  humanizeBinaryBytes,
+  LoadingInline,
+  ResourceIcon,
+  withHandlePromise,
+} from '@console/internal/components/utils/index';
+import {
+  createModalLauncher,
+  ModalBody,
+  ModalSubmitFooter,
+  ModalTitle,
+  ModalComponentProps,
+} from '@console/internal/components/factory';
+import { k8sCreate, K8sResourceKind } from '@console/internal/module/k8s';
+import { PersistentVolumeClaimModel } from '@console/internal/models/index';
+import { HandlePromiseProps } from '@console/internal/components/utils/promise-component';
+import { usePrometheusPoll } from '@console/internal/components/graphs/prometheus-poll-hook';
+import { PrometheusEndpoint } from '@console/internal/components/graphs/helpers';
+import { getInstantVectorStats } from '@console/internal/components/graphs/utils';
+import { DataPoint, PrometheusResponse } from '@console/internal/components/graphs/index';
+import { getPVCUsedCapacityQuery } from '../../../constants/queries';
+import './_clone-pvc-modal.scss';
+
+const accessModeRadios = Object.freeze({
+  ReadWriteOnce: 'Single User (RWO)',
+  ReadWriteMany: 'Shared Access (RWX)',
+  ReadOnlyMany: 'Read Only (ROX)',
+});
+
+const ClonePVCModal = withHandlePromise((props: ClonePVCModalProps) => {
+  const { close, cancel, resource, handlePromise, errorMessage, inProgress } = props;
+  const pvcName: string = _.get(resource, 'metadata.name');
+  const [buttonDisabled, setButton] = React.useState(false);
+  const [clonePVCName, setClonePVCName] = React.useState(`${pvcName}-clone`);
+
+  const query: string = getPVCUsedCapacityQuery(pvcName);
+
+  const [response, error, loading] = usePrometheusPoll({
+    endpoint: PrometheusEndpoint.QUERY,
+    query,
+  }) as [PrometheusResponse, Error, boolean];
+
+  const results: DataPoint[] = getInstantVectorStats(response, null, humanizeBinaryBytes);
+  const pvcUsedCapacity: string = _.get(results[0], 'label')
+    ? _.get(results[0], 'label')
+    : 'No Data';
+
+  const submit = (event: React.FormEvent<EventTarget>) => {
+    event.preventDefault();
+
+    const pvcCloneObj = {
+      apiVersion: 'v1',
```

Comment:
Consistently use the values from the `PersistentVolumeClaimModel` if you're going to use it.

```suggestion
      apiVersion: PersistentVolumeClaimModel.apiVersion,
```